src/Recursor.h

```diff
@@ -320,7 +320,7 @@ void Recursor<Derived>::FillBeta(const M& guide, M& beta) const
 
     // Recursively calculate [Probability transition to next state] *
     // [Probability of emission at that state] * [Probability from that state]
-    for (int j = J - 1; j > 0; --j) {
+    for (size_t j = J - 1; j > 0; --j) {
```

Comment:
Though an assert(J > 0) would probably be good (up above, amongst the other asserts). Good catch @pb-cdunn